run-docker.sh

```diff
@@ -1,17 +1,22 @@
+# Exit on failure
+set -e
+
+# Build on-chain programs first
+anchor build
+cp programs/target/deploy/gummyroll.so docker-vol/gummyroll.so
+cp programs/target/deploy/gummyroll_crud.so docker-vol/gummyroll_crud.so
+cp programs/target/deploy/bubblegum.so docker-vol/bubblegum.so
+
 # Off Chain Setup
- # Speed up by using cargo remote
-cross build --target x86_64-unknown-linux-gnu --package nft_api
```

Comment:
I think this will break people trying to build on Mac, because even with correct target, I think cargo build will not work because some dependencies need to compile with x86_64-unknown-linux version of gcc (and not clang).

I am fine with this change because I encountered major problems when trying to get the composed docker container to run properly on M1 Mac, and I doubt anyone else wants to debug that on M1 Mac either.  I switched to building and running this on Linux.

Just mentioning this as an FYI in case Mac compatibility is going to be a requirement for this project.

Reply:
I think we are fine without Mac compatibility. This should be running in cloud anyways